### PR TITLE
Revert clamping scroll simulation changes

### LINKED ISSUE #83632: Scroll view regains velocity as it scrolls
## Steps to Reproduce

Open and scroll license details in LicensePage

The issue is not present in the current gallery build, i.e. this is a regression

**Expected results:**

https://user-images.githubusercontent.com/39104740/120113922-6983f980-c185-11eb-92f6-7269e84f1ba2.mp4

**Actual results:**

https://user-images.githubusercontent.com/39104740/120113846-0befad00-c185-11eb-875b-9cdf2e8ffcc2.mp4

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(App());
}

class App extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'App',
      home: Scaffold(
        body: Center(
          child: Builder(
            builder: (context) {
              return ElevatedButton(
                child: Text('f'),
                onPressed: () {
                  showLicensePage(context: context);
                },
              );
            }
          )
        ),
      ),
    );
  }
}
```
</details>

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 2.3.0-17.0.pre.120, on Microsoft Windows [Version 10.0.19041.985], locale ru-RU)
    • Flutter version 2.3.0-17.0.pre.120 at C:\dev\src\flutter
    • Upstream repository git@github.com:nt4f04und/flutter.git
    • Framework revision c42ec8592f (2 days ago), 2021-05-28 22:34:02 -0400
    • Engine revision aca8aa2335
    • Dart version 2.14.0 (build 2.14.0-165.0.dev)

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.7)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30621.155
    • Windows 10 SDK version 10.0.19041.0

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] IntelliJ IDEA Community Edition (version 2020.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2020.3.3
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.56.2)
    • VS Code at C:\Users\danya\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.22.0

[✓] Connected device (4 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)       • windows       • windows-x64    • Microsoft Windows [Version 10.0.19041.985]
    • Chrome (web)            • chrome        • web-javascript • Google Chrome 90.0.4430.212
    • Edge (web)              • edge          • web-javascript • Microsoft Edge 90.0.818.66

! Doctor found issues in 1 category.
```
</details>

### PR DESCRIPTION
**Pull request by Piinks**
This reverts https://github.com/flutter/flutter/pull/77497 which sought to address https://github.com/flutter/flutter/issues/16371
The original change caused a regression in https://github.com/flutter/flutter/issues/83632
Fixes https://github.com/flutter/flutter/issues/83632

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/widgets/scrollable_fling_test.dart [hunk 1]
```diff
@@ -44,6 +44,11 @@ void main() {
     expect(getCurrentOffset(), dragOffset);
     await tester.pump(const Duration(seconds: 5));
     final double androidResult = getCurrentOffset();
+    // Regression test for https://github.com/flutter/flutter/issues/83632
+    // Before changing these values, ensure the fling results in a distance that
+    // makes sense. See issue for more context.
+    expect(androidResult, greaterThan(394.0));
+    expect(androidResult, lessThan(395.0));
```
Comment (Piinks) [inline packages/flutter/test/widgets/scrollable_fling_test.dart:51]: I added this to draw attention to any future regressions. The rest of the PR is a straight revert from the original.

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
We should probably cherry-pick this into the latest release.

**Comment by Piinks** [comment]
I agree, I'll label the issue for cp once this lands

**Pull request merged**